### PR TITLE
Dev/william/conn close when owner dead

### PR DESCRIPTION
**Pull request by qzhuyan**

### CODE REVIEW

File: c_src/quicer_ctx.c [hunk 1]
```diff
@@ -57,6 +57,7 @@ init_c_ctx()
 void
 destroy_c_ctx(QuicerConnCTX *c_ctx)
 {
+  enif_demonitor_process(c_ctx->env, c_ctx, &c_ctx->owner_mon);
   enif_release_resource(c_ctx);
```
Comment (qzhuyan) [inline c_src/quicer_ctx.c:61]: add note that release resource is aync call

File: c_src/quicer_nif.c [hunk 2]
```diff
@@ -513,42 +513,55 @@ resource_listener_down_callback(__unused_parm__ ErlNifEnv *caller_env,
...
-  // todo
+  QuicerConnCTX *c_ctx = ctx;
+  if (!ctx)
+    {
+      return;
+    }
+  else
+    {
+      TP_CB_3(start, c_ctx->Connection, (uint64_t)ctx);
+      MsQuic->ConnectionShutdown(
```
Comment (qzhuyan) [inline c_src/quicer_nif.c:544]: ConnectionShutdown returns void, no way to check if the shutdown is success or not.

File: c_src/quicer_stream.c [hunk 3]
```diff
@@ -593,26 +594,36 @@ recv2(ErlNifEnv *env, __unused_parm__ int argc, const ERL_NIF_TERM argv[])
...
   ERL_NIF_TERM ret = ATOM_OK;
   QuicerStreamCTX *s_ctx;
+  uint32_t app_errcode = 0, flags = 0;
   if (!enif_get_resource(env, argv[0], ctx_stream_t, (void **)&s_ctx))
     {
       return ERROR_TUPLE_2(ATOM_BADARG);
     }
+
+  if (!enif_get_uint(env, argv[1], &flags))
+    {
```
Comment (qzhuyan) [inline c_src/quicer_stream.c:611]: add notes why we don't check the flags

File: src/quicer.erl [hunk 4]
```diff
@@ -154,9 +160,16 @@ accept(LSock, Opts, Timeout) ->
 close_connection(Conn) ->
   close_connection(Conn, 5000).
 
--spec close_connection(connection_handler(), timer:timeout()) -> ok.
 close_connection(Conn, Timeout) ->
```
Comment (qzhuyan) [inline src/quicer.erl:163]: missing spec.

**Review by k32 — APPROVED** [review]

**Pull request merged**